### PR TITLE
Add model Rembert

### PR DESCRIPTION
**Pull request by Beacontownfc**
Description
Add new model RemBert
The model weight:
链接：https://aistudio.baidu.com/aistudio/datasetdetail/129105

### CODE REVIEW

File: paddlenlp/transformers/rembert/tokenizer.py [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+            pad_token="[PAD]",
+            mask_token="[MASK]", ):
+
+        self.do_lower_case = do_lower_case
+        self.remove_space = remove_space
+        self.keep_accents = keep_accents
+        self.vocab_file = vocab_file
+        self.sp_model = spm.SentencePieceProcessor()
+        self.sp_model.Load(vocab_file)
+        self.cls_token = cls_token
```
Comment (gongel) [inline paddlenlp/transformers/rembert/tokenizer.py]: 101-105行可以省略

File: paddlenlp/transformers/rembert/modeling.py [hunk 2]
```diff
@@ -0,0 +1,1014 @@
...
+                                         key_layer.transpose((0, 1, 3, 2)))
+
+        attention_scores = attention_scores / math.sqrt(
+            self.attention_head_size)
+        if attention_mask is not None:
+            # Apply the attention mask is (precomputed for all layers in RemBertModel forward() function)
+            attention_scores = attention_scores + attention_mask
+
+        # Normalize the attention scores to probabilities.
+        attention_probs = nn.Softmax(axis=-1)(attention_scores)
```
Comment (gongel) [inline paddlenlp/transformers/rembert/modeling.py]: 这里可以挪到__init__或者F.softmax

File: paddlenlp/transformers/rembert/modeling.py [hunk 3]
```diff
@@ -0,0 +1,1014 @@
...
+    "gelu": F.gelu,
+    "gelu_new": gelu_new,
+    "tanh": F.tanh,
+    "sigmoid": F.sigmoid,
+    "mish": mish,
+    "linear": linear_act,
+    "swish": swish,
+}
+
+layer_norm_eps = 1e-12
```
Comment (gongel) [inline paddlenlp/transformers/rembert/modeling.py]: 这个可以尝试挪到config里面，便于配置

File: paddlenlp/transformers/rembert/modeling.py [hunk 4]
```diff
@@ -0,0 +1,1014 @@
...
+            An instance of RemBertModel.
+        num_classes (int, optional):
+            The number of classes.
+    """
+
+    def __init__(self, rembert, num_classes):
+        super(RemBertForSequenceClassification, self).__init__()
+        self.rembert = rembert
+        self.dense = nn.Linear(self.rembert.config['hidden_size'], num_classes)
+        self.loss_fn = nn.CrossEntropyLoss()
```
Comment (gongel) [inline paddlenlp/transformers/rembert/modeling.py]: 这个删掉吧

**Review by gongel — CHANGES_REQUESTED** [review]

**Comment by gongel** [comment]
感谢贡献，麻烦根据comment修改下哈😊 @Beacontownfc

File: paddlenlp/transformers/rembert/tokenizer.py (hunk 1 above)
Comment (Beacontownfc) [reply paddlenlp/transformers/rembert/tokenizer.py]: Done

File: paddlenlp/transformers/rembert/modeling.py (hunk 2 above)
Comment (Beacontownfc) [reply paddlenlp/transformers/rembert/modeling.py]: Done

File: paddlenlp/transformers/rembert/modeling.py (hunk 3 above)
Comment (Beacontownfc) [reply paddlenlp/transformers/rembert/modeling.py]: Done

File: paddlenlp/transformers/rembert/modeling.py (hunk 4 above)
Comment (Beacontownfc) [reply paddlenlp/transformers/rembert/modeling.py]: Done

**Comment by Beacontownfc** [comment]
已根据您的要求进行了修改 @gongel

**Review by gongel — DISMISSED** [review]
LGTM

**Comment by Beacontownfc** [comment]
@yingyibiao 一切OK，请求批准合入😊

**Comment by gongel** [comment]
@Beacontownfc padding输入有问题，麻烦check一下哈
```python
import io
import os
import shutil
import importlib

import numpy as np
import paddle
import torch
import transformers as hfnlp
import paddlenlp
from paddlenlp.data import Pad
import paddlenlp.transformers as ppnlp

os.environ["TRANSFORMERS_CACHE"] = "./hf/"
os.environ["PPNLP_HOME"] = "./pdnlp/"


def compute_diff(torch_data, paddle_data):
	torch_data = torch_data.detach().numpy()
	paddle_data = paddle_data.numpy()
	out_dict = dict()
	diff = np.abs(torch_data - paddle_data)
	out_dict = "max: {}    mean: {}    min: {}".format(diff.max(), diff.mean(), diff.min())
	return out_dict


def compare_base(model_id):
	sentences = [
		"This is an example sentence.", 
		"Each sentence is converted .", 
		"欢迎使用 PaddlePaddle  。",
		"欢迎使用 PaddleNLP 。"
	]
	
	# Calculate HF output
	hf_tokenizer = hfnlp.RemBertTokenizer.from_pretrained('google/rembert') # google/rembert
	hf_model = hfnlp.RemBertModel.from_pretrained('google/rembert') # google/rembert
	hf_model.eval()
	with torch.no_grad():
		hf_inputs = hf_tokenizer(sentences, padding=True, return_tensors="pt")
		print(hf_inputs)
		hf_out = hf_model(**hf_inputs).last_hidden_state
	
	# Calculate Paddle output
	pd_tokenizer = ppnlp.RemBertTokenizer.from_pretrained('rembert')
	pd_model = ppnlp.RemBertModel.from_pretrained('rembert')
	pd_model.eval()
	with paddle.no_grad():
		pd_inputs = pd_tokenizer(sentences)
		input_ids = paddle.to_tensor(Pad(axis=0, pad_val=pd_tokenizer.pad_token_id)([pd_input for pd_input in pd_inputs["input_ids"]]))
		token_type_ids = paddle.to_tensor(Pad(axis=0, pad_val=pd_tokenizer.pad_token_type_id)([pd_input for pd_input in pd_inputs["token_type_ids"]]))
		print(input_ids)
		print(token_type_ids)
		pd_out = pd_model(input_ids, token_type_ids)[0]

	return compute_diff(hf_out, pd_out)

print(compare_base("rembert"))
```

**Comment by Beacontownfc** [comment]
> ```python
> input_ids = paddle.to_tensor(Pad(axis=0, pad_val=pd_tokenizer.pad_token_id)([pd_input for pd_input in pd_inputs["input_ids"]]))
> ```

您好，这行代码
`input_ids = paddle.to_tensor(Pad(axis=0, pad_val=pd_tokenizer.pad_token_id)([pd_input for pd_input in pd_inputs["input_ids"]]))`
换成BertTokenizer运行也是出现报错，我认为此行代码应该改成这样
`input_ids = paddle.to_tensor( Pad(axis=0, pad_val=pd_tokenizer.pad_token_id)([pd_input["input_ids"] for pd_input in pd_inputs]))`
这样代码正常运行了

**Comment by gongel** [comment]
请问你的paddlenlp是哪个版本呢？develop代码目前没问题

**Review by gongel — APPROVED** [review]

**Comment by gongel** [comment]
@Beacontownfc 可以再适配一下[Auto](https://github.com/PaddlePaddle/PaddleNLP/blob/develop/paddlenlp/transformers/auto)哈
